File: R/R/diagarray.R
diagarray <- function(S, i=1, out=c()){
  #' Compute matrix of diagonals
  #' 
  #' Creates a matrix of diagonals from an array of square matrices
  #' 
  #' @param S tri-dimensional array with pages formed by square matrices
  #' 
  #' @return matrix whose columns are the main diagonals of the pages of S
  #' @export
  if (i <= dim(S)[3])
    out <- diagarray(S, i+1 , out=cbind(out, diag(S[,,i])))
  return(out)
}

File: matlab/Utilities/diagarray.m
function [Out] = diagarray (S)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Creates a matrix of diagonals from an array of square matrices
% INPUT:
% S:    tri-dimensional array with pages formed by square matrices
% OUTPUT:
% Out:  matrix whose columns are the main diagonals of the pages of S
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Out = zeros (size(S,1),size(S,3));
for i = 1:size(S,3)
    Out(:,i) = diag(S(:,:,i));
end
